src/groovy/org/transmartproject/db/dataquery/highdim/snp_lz/SnpLzModule.groovy

```diff
@@ -114,6 +116,12 @@ class SnpLzModule extends AbstractHighDimensionDataTypeModule {
         [standardDataConstraintFactory,
          chromosomeSegmentConstraintFactory,
          new MapBasedParameterFactory(
+                 (TRIALNAME_CONSTRAINT_NAME): { Map<String, Object> params ->
+                     validateParameterNames(['trialName'], params)
+                     new PropertyDataConstraint(
+                             property: 'snp.trialName',
+                             values:    params.trialName)
+                 },
```

Comment:
If I understand correctly the problem is that (unlike with the other data types), we don't join the data type with subject_sample_mapping and therefore applying the trial restriction as an assay constraint is not enough. Is this correct?

If so, the user still shouldn't worry about this, as the constraint to `snp.trialName` can be applied automatically by looking at the trials of the assays.

Plus, this code is not exercised by any test.